File: 2015/day02.py
lines = open("input.txt").read().strip().splitlines()

part1 = 0
part2 = 0

for line in lines:
    l, w, h = map(int, line.split("x"))
    part1 += 2*l*w + 2*w*h + 2*h*l + min(l*w, w*h, h*l)
    part2 += 2*min(l+w, w+h, h+l) + l*w*h

print(part1)
print(part2)

File: 2015/day02.js
const lines = require("fs").readFileSync("input.txt", "utf8").trim().split("\n");

const part1 = lines.map(line => {
  const [l, w, h] = line.split("x")
  const smallest = Math.min(l*w, w*h, h*l);
  return 2*l*w + 2*w*h + 2*h*l + smallest;
}).reduce((curr, acc) => curr + acc, 0);

console.log(part1);
